src/main/webapp/app/assessment/assessment-header/assessment-header.component.ts

```diff
@@ -93,11 +93,19 @@ export class AssessmentHeaderComponent {
         // if there is no 'save' button
         if (this.result?.completionDate) {
             return true;
+        } else if (Result.hasNonEmptyAssessmentNote(this.result)) {
+            // this is almost identical to submitDisabled, but without the assessmentsAreValid check
+            // otherwise, we wouldn't be able to save the assessment note without making prior changes to the feedback
+            return !this.isAssessor || this.saveBusy || this.submitBusy || this.cancelBusy;
         } else {
-            return !this.assessmentsAreValid || !this.isAssessor || this.saveBusy || this.submitBusy || this.cancelBusy;
+            return this.saveDisabledWhenNoAssessmentNotePresent;
```

Comment:
Can you move this check into a method (e.g. `saveDisabledWithAssessmentNode`) and then change `saveDisabledWhenNoAssessmentNotePresent` to just `this.saveDisabledWithAssessmentNode || !this.assessmentsAreValid`?

Reply:
Done.